### ISSUE TITLE
Home screen logo and private browsing button colors should be adapted to wallpapers.

### ISSUE THREAD

**Issue by mcarare** [mozilla-mobile/fenix#26644]
When using certain wallpapers, the Firefox logo and the PB button are not very visible. 

We should have the ability to adapt colors in order to provide a good contrast ratio.

cc: @Kate-Galetski

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-21392)

**Comment by delia-pop** [comment]
We tested this on 107.0a1 from 09/22 with Xiaomi Pad5 (Android 11), Google Pixel 6 (Android 13), Xiaomi 12 Pro (Android 12), Lenovo Yoga Tab 11 (Android 11), LG G7 fit (Android 8.1.0) and One Plus 9 PRO (Android 12). The following were observed:

- on all devices: When returning from Private Browsing mode, the PB icon is improperly displayed for a split second as a solid white/dark circle depending on the theme used, but only when a wallpaper it set.

https://user-images.githubusercontent.com/89388888/191740827-e9ce79f3-645d-44b7-b64d-ce0684fa5756.mp4

- on Xiaomi Pad5 (Android 11) (which might be prone to color issues) : When entering Private Browsing mode, the PB icon is displayed as a solid white/dark circle depending on the theme used, regardless of whether a wallpaper is set or not.

https://user-images.githubusercontent.com/89388888/191740835-4d869b8c-5ffc-45e5-89db-4a7260e6d2e7.mp4

**Comment by Mugurell** [comment]
Interesting issue.
Going by the first video I see 3 stages for when switching to normal mode which I think show the problem:

| private mode | in-between | normal mode|
| --- | --- | --- |
| <img src="https://user-images.githubusercontent.com/11428869/191745158-fe418999-1271-4c56-9a85-65af734129c1.png"> | <img src="https://user-images.githubusercontent.com/11428869/191745316-aeaa2d17-b58b-4fe0-9f07-23fe11e5da9e.png"> | <img src="https://user-images.githubusercontent.com/11428869/191746126-c6261cb1-e799-4c21-b777-79003c77f468.png"> |

So there is an overarching issue: slow transition that affects not only the private browsing button.
And then the issue with the private browsing button is the result of the race stemming from the above issue between:
- the background which stays the same until private mode is fully off - in the middle image we see the toolbar still in private mode
- the new tint that is applied earlier - in the middle image we see the normal mode wallpaper - which will apply the it's `textColor` to the logo and private browsing button.

So as a solution here we should also change for the private browsing button both the background and the image to ensure consistency.

**Comment by delia-pop** [comment]
Verified as fixed on Beta 106.0b3 as well, with Lenovo Yoga Tab 11 (Android 11).
